lib/grammar-registry.js

```diff
@@ -1,273 +1,324 @@
+  //   * `grammar` {Grammar} that was updated.
+  //
+  // Returns a {Disposable} on which `.dispose()` can be called to unsubscribe.
+  onDidUpdateGrammar(callback) {
+    return this.emitter.on('did-update-grammar', callback);
+  }
 
+  // Public: Invoke the given callback when a grammar is removed from the registry.
+  //
+  // * `callback` {Function} to call when a grammar is removed.
+  //   * `grammar` {Grammar} that was removed.
+  //
+  // Returns a {Disposable} on which `.dispose()` can be called to unsubscribe.
+  onDidRemoveGrammar(callback) {
+    return this.emitter.on('did-remove-grammar', callback);
+  }
 
-    /*
-    Section: Managing Grammars
-     */
+  // ###
+  // Section: Managing Grammars
+  // ###
 
-    GrammarRegistry.prototype.getGrammars = function() {
-      return _.clone(this.grammars);
-    };
+  // Public: Get all the grammars in this registry.
+  //
+  // Returns a non-empty {Array} of {Grammar} instances.
+  getGrammars() {
+    return _.clone(this.grammars);
+  }
 
-    GrammarRegistry.prototype.grammarForScopeName = function(scopeName) {
-      return this.grammarsByScopeName[scopeName];
-    };
+  // Public: Get a grammar with the given scope name.
+  //
+  // * `scopeName` A {String} such as `"source.js"`.
+  //
+  // Returns a {Grammar} or undefined.
+  grammarForScopeName(scopeName) {
+    return this.grammarsByScopeName[scopeName];
+  }
 
-    GrammarRegistry.prototype.addGrammar = function(grammar) {
-      this.grammars.push(grammar);
-      this.grammarsByScopeName[grammar.scopeName] = grammar;
-      if (grammar.injectionSelector != null) {
-        this.injectionGrammars.push(grammar);
-      }
-      this.grammarUpdated(grammar.scopeName);
-      if (Grammar.includeDeprecatedAPIs) {
-        this.emit('grammar-added', grammar);
-      }
-      this.emitter.emit('did-add-grammar', grammar);
-      return new Disposable((function(_this) {
-        return function() {
-          return _this.removeGrammar(grammar);
-        };
-      })(this));
-    };
+  // Public: Add a grammar to this registry.
+  //
+  // A 'grammar-added' event is emitted after the grammar is added.
+  //
+  // * `grammar` The {Grammar} to add. This should be a value previously returned
+  //   from {::readGrammar} or {::readGrammarSync}.
+  //
+  // Returns a {Disposable} on which `.dispose()` can be called to remove the
+  // grammar.
+  addGrammar(grammar) {
+    this.grammars.push(grammar);
+    this.grammarsByScopeName[grammar.scopeName] = grammar;
+    if (grammar.injectionSelector != null) {
+      this.injectionGrammars.push(grammar);
+    }
+    this.grammarUpdated(grammar.scopeName);
+    if (Grammar.includeDeprecatedAPIs) {
+      this.emit('grammar-added', grammar);
+    }
+    this.emitter.emit('did-add-grammar', grammar);
+    return new Disposable(((_this) => {
+      return function() {
+        return _this.removeGrammar(grammar);
+      };
+    })(this));
```

Comment:
This should be the same:
```js
return new Disposable(() => 
  this.removeGrammar(grammar)
);
```

Reply:
You totally could, but I'll test to see if that works. Thanks!